docs(idempotency): add callout about undefined early return (#3055)
aws-powertools/powertools-lambda-typescript: docs/utilities/idempotency.md

    --8<-- "examples/snippets/idempotency/types.ts:3:16"
    ```

> Note that for the middleware to work, your Lambda function handler must return a value different from `undefined`. This is a [known limitation of Middy.js early return feature](https://github.com/middyjs/middy/issues/1236). If your use case requires early returns, you can use the `makeIdempotent` function wrapper instead.

Reviewer: ```suggestion
For the middleware to work, your Lambda function handler must return a value different from `undefined`. This is a [known limitation of the early return feature in Middy.js](https://github.com/middyjs/middy/issues/1236). If your use case requires early returns, you can use the `makeIdempotent` function wrapper instead.
```
